bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -64,13 +65,20 @@ export class StringFormatter extends CellFormatter {
       text_align: [ p.TextAlignSpec, {value: "left"} ],
       text_color: [ p.ColorSpec, null ],
       background_color: [ p.ColorSpec, null ],
-      nan_format: [ Str, "-"],
+      nan_format: [ Str, "NaN"],
+      null_format: [ Str, "(null)"],
     }))
   }
 
   override doFormat(_row: any, _cell: any, value: any, _columnDef: any, dataContext: any): string {
     const {font_style, text_align, text_color, background_color} = this
 
+    if (Number.isNaN(value)) {
+      value = this.nan_format
+    } else if (value == null) {
+      value = this.null_format
+    }
```

Comment:
FWIW I think this is slightly awkward with the other formatters inheriting from `StringFormatter`, since they also do the same checks and then often call `super.doFormat`. 

That said, things already seem a bit vague here: e.g. is ` StringFormatter a formatter *for string values* or a formatter that formats anything, *to string values*? In actuality it seems more like the name `TextPropertiesFormatter` would be more descriptive here.